### PR TITLE
Added new muxorder tests

### PR DESCRIPTION
**Pull request by williamzhu17**
New tests for muxorder pass
- Tests all operators
- Tests nested muxes

----

> [!IMPORTANT]
> Added comprehensive tests for mux operations in `muxorder.ys`, covering various operators, nested conditions, and edge cases.
> 
>   - **Tests Added**:
>     - Added tests for basic mux operations with different operators: `+`, `*`, `&`, `|`, `^`, `~^`.
>     - Added tests for nested mux operations.
>     - Added tests for scenarios with uneven operand widths.
>   - **Behavior**:
>     - Tests ensure correct transformation and rewiring of operations in mux patterns.
>     - Includes assertions to verify correct rewiring of adders, multipliers, and other operators.
>     - Tests cover cases where transformations should not occur due to specific conditions (e.g., more fanouts, width mismatches).
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=Silimate%2Fyosys&utm_source=github&utm_medium=referral)<sup> for 82fa68aa2d92e0f68415163221543fffa6b7bc1b. You can [customize](https://app.ellipsis.dev/Silimate/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: tests/peepopt/muxorder.ys [hunk 1]
```diff
@@ -376,3 +374,148 @@ opt_clean
...
+log -push
+design -reset
+read_verilog <<EOF
+module top(a, b, s, y);
+	input wire a;
+	input wire b;
+	output wire y;
+	input wire s;
+	assign y = s ? a | b : a;
+endmodule
```
[the remark this answers is not in the document]
Comment (williamzhu17) [reply tests/peepopt/muxorder.ys:449]: other tests use bitwise too

File: tests/peepopt/muxorder.ys [hunk 2]
```diff
@@ -376,3 +374,148 @@ opt_clean
 equiv_opt -assert peepopt -muxorder
 design -load postopt
 select -assert-any t:$add %co1 %a w:y %i # assert adder rewired
+
+log -pop
+log -header "Test transform when widths are uneven with no intermediate values"
+log -push
+design -reset
```
[the remark this answers is not in the document]
Comment (akashlevy) [reply tests/peepopt/muxorder.ys:381]: No
